app/views/layouts/application.html.erb

```diff
@@ -2,13 +2,26 @@
 <html>
 <head>
   <title>Blocipedia</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
+  <div class="container">
+    <ul class="nav nav-tabs">
+      <li><%= link_to "Blocipedia", welcome_index_path %></li>
+    </ul>
+    <% if user_signed_in? %>
+      Signed in as: <strong><%= current_user.email %></strong> |
+      <%= link_to 'Edit profile', edit_user_registration_path, :class => 'navbar-link' %> - <%= link_to "Sign Out", destroy_user_session_path, method: :delete, :class => 'navbar-link'  %>
+    <% else %>
+      <%= link_to "Sign Up", new_user_registration_path, :class => 'navbar-link'  %> -
+      <%= link_to "Sign In", new_user_session_path, :class => 'navbar-link'  %>
+    <% end %>
```

Comment:
Have you thought about adding error handling yet to fulfill these two scenarios:  What happens if you attempt to sign up with an invalid email?  What happens if you attempt to sign up with a duplicate email?

You can simply use the flash messages here to display validation errors to the user (Similar to how you did in bloccit).